### PR TITLE
Feature/mediastream

### PR DESCRIPTION
**Pull request by sijumoncy**
#377 

- Content Type for medias ( Video, Audio, Image)
- Header Range Error when request as normal api call ( Not From Player or stream supported methods)
- Manual Check Done for Stream and Download apis for Video, Audio and Image
- Auth verified as per rule ( Manually )

### CODE REVIEW

File: app/crud/media_crud.py [hunk 1]
```diff
@@ -25,22 +27,26 @@ def get_gitlab_stream(request, repo, tag, file_path,permanent_link,**kwargs):#py
...
-        content_type =  "video/mp4"
-    elif url.endswith("mov") or url.endswith("MOV"):
-        content_type =  "video/quicktime"
-    else:
-        raise Exception("Unsupported video format!")
+    content_type = mimetypes.guess_type(url.split("/")[-1], strict=True)
+    if content_type is None:
+        raise Exception("Unsupported media format!")
+
+    if "image" not in content_type[0]:
```
Comment (kavitharaju) [inline app/crud/media_crud.py]: Hmm.. now I wonder if image needs to be brought under this API...

File: app/crud/media_crud.py [hunk 2]
```diff
@@ -25,22 +27,26 @@ def get_gitlab_stream(request, repo, tag, file_path,permanent_link,**kwargs):#py
...
     else:
         url = permanent_link
 
-    if url.endswith("mp4"):
-        content_type =  "video/mp4"
-    elif url.endswith("mov") or url.endswith("MOV"):
-        content_type =  "video/quicktime"
-    else:
-        raise Exception("Unsupported video format!")
+    content_type = mimetypes.guess_type(url.split("/")[-1], strict=True)
```
Comment (kavitharaju) [inline app/crud/media_crud.py:38]: This is the better way to do it

**Review by kavitharaju — APPROVED** [review]
The implemented changes looks good. 
There was a mention of including gitlabrepo as a contnettype in the seed DB in the issue. Is that done? Will it be reflected upon a fresh DB creation?

What is the auth congiuration now? Can any API-user(or unregistered user) access the API, when the content is marked as open-access?

**Comment by sijumoncy** [comment]
- Removed Image from stream ( Download will support it . Manually tested)
- Auth test done manually ( Api user access open-access content)
- test cases updated for errors 
- mimetypes is used based on most of the suggestions found  to work with media content type in python

**Review by kavitharaju — APPROVED** [review]

**Pull request merged**